i3lock.c

```diff
@@ -1132,6 +1151,131 @@ static const struct raw_pixel_format raw_fmt_bgr = {3, 2, 1, 0};
+        if (cairo_surface_status(gif_img[idx].img) != CAIRO_STATUS_SUCCESS) {
+            fprintf(stderr, "Could not create surface: %s\n",
+                    cairo_status_to_string(cairo_surface_status(gif_img[idx].img)));
+            free(gif_img);
+            gif_img = NULL;
+            goto read_gif_image_clean;
+        }
+        cairo_surface_flush(gif_img[idx].img);
+        uint32_t *data = (uint32_t *)cairo_image_surface_get_data(gif_img[idx].img);
+
+        // TODO: Chek L,R,W,H && color map?
+        /* Find Graphics Control extension and delay */
+        for (int iext = 0; iext < pimg->ExtensionBlockCount; ++iext) {
+            ExtensionBlock *pext = pimg->ExtensionBlocks + iext;
+            switch (pext->Function) {
+                case GRAPHICS_EXT_FUNC_CODE:
+                    DGifExtensionToGCB(pext->ByteCount, pext->Bytes, &gc);
+                    gif_img[idx].delay_sec = gc.DelayTime*0.01;
+                    break;
+                default:
+                    gif_img[idx].delay_sec = 0;
+            }
+        }
+
+        /* Set general image attributes */
+        ColorMapObject *cmap_img = pimg->ImageDesc.ColorMap;
+        if (!cmap_img) cmap_img = cmap;
+        int img_left = pimg->ImageDesc.Left;
+        int img_width = pimg->ImageDesc.Width;
+        int img_right = img_left + img_width;
+        int img_top = pimg->ImageDesc.Top;
+        int img_bottom = img_top + pimg->ImageDesc.Height;
+
+        /* Handle disposal mode */
+        int data_size = width*height*((int)sizeof(width));
+        switch (gc.DisposalMode) {
+            case DISPOSE_DO_NOT:
+                if (data_prev) {
+                    memcpy(data, data_prev, data_size);
+                } else {
+                    memset(data, 0, data_size);
+                }
+                break;
+            case DISPOSE_BACKGROUND:
+                memset(data, bg_color, data_size);
+                break;
+            default:
+                memset(data, 0, data_size);
+        }
+
+        /* Read RGB */
+        for (int x = 0; x < width; ++x) {
+            for (int y = 0; y < height; ++y) {
+                int color_idx;
+                if (x < img_left || x >= img_right || y < img_top || y >= img_bottom) {
+                    color_idx = bg_idx;
+                } else {
+                    color_idx = *(pimg->RasterBits + (x - img_left) + ((y - img_top)*img_width));
+                }
+
+                if (color_idx != gc.TransparentColor) {
+                    GifColorType *cmap_rgb = cmap_img->Colors + color_idx;
+                    uint32_t rgb = 0 | cmap_rgb->Blue | cmap_rgb->Green<<8 | cmap_rgb->Red<<16;
+                    data[x + (y*width)] = rgb;
+                }
+            }
+        }
+        data_prev = data;
+
+        cairo_surface_mark_dirty(gif_img[idx].img);
+    }
+
+read_gif_image_clean:
+    if (!DGifCloseFile(gif, &err)) {
+        fprintf(stderr, "DGifCloseFile failed with err=%d\n", err);
```

Comment:
I think this can be a `DEBUG` call instead, with string: `DGifCloseFile call failed (Error %d)\n`